adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -510,12 +560,30 @@ case class ReferenceRegion(
   }
 
   override def hashCode: Int = {
-    var result = 37
-    result = 41 * result + (if (referenceName != null) referenceName.hashCode else 0)
-    result = 41 * result + start.hashCode
-    result = 41 * result + end.hashCode
-    result = 41 * result + (if (strand != null) strand.ordinal() else 0)
-    result
+    // add 37 here because this is the start of the hashCode
+    val nameHashCode = 37 + {
```

Comment:
http://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#hash-java.lang.Object...- handles all of this reasonably in a one-liner.

If performance is a concern, and if ReferenceRegion is immutable, the hash code could be computed in the ctr and assigned to a final `Int` field.